### PR TITLE
Dram stack

### PR DESCRIPTION
**Pull request by skeqiqevian**
Need to support `DRAM_STACK` in addition to `DRAM_STATIC` because of parallel semantics. If we want to allocate temporary memory buffers in parallel code, `static` memory will lead to races because it isn't thread local.

Depends on #526

### CODE REVIEW

File: src/exo/LoopIR_scheduling.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 from .new_eff import (
     SchedulingError,
     Check_ReorderStmts,
+    Check_ParallelizeLoop,
```
Comment (yamaguchi1024) [inline src/exo/LoopIR_scheduling.py:21]: This is unnecessary, right?

File: src/exo/LoopIR_compiler.py [hunk 2]
```diff
@@ -397,10 +397,10 @@ def from_lines(x):
         else:
             is_public_decl = id(p) in orig_procs
 
+            p = ParallelAnalysis().run(p)
```
Comment (yamaguchi1024) [inline src/exo/LoopIR_compiler.py:400]: Why?

Comment (skeqiqevian) [reply src/exo/LoopIR_compiler.py:400]: As in why is it first? Or why at all?

Comment (yamaguchi1024) [reply src/exo/LoopIR_compiler.py:400]: Why did you move it to the first?

Comment (skeqiqevian) [reply src/exo/LoopIR_compiler.py:400]: I don't think the current effects analysis handle frees correctly, so I moved it first. We'll probably have to revisit this ordering later for GPUs, but in the current state, I don't think the order particularly matters?

Comment (yamaguchi1024) [reply src/exo/LoopIR_compiler.py:400]: Okay yeah that's fine

File: src/exo/LoopIR_scheduling.py (hunk 1 above)
Comment (yamaguchi1024) [reply src/exo/LoopIR_scheduling.py:21]: Do you use Check_ParallelizeLoop in LoopIR_scheduling.py? If so we shouldn't right?

**Review by yamaguchi1024 — APPROVED** [review]

**Pull request merged**